### PR TITLE
Adding common functions in helper folder

### PR DESCRIPTION
**Pull request by ramreddya**

### CODE REVIEW

File: sriracha/helper/manage_user_groups.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    try:
+        pwd.getpwnam(username)
+        return True
+    except:  # pylint: disable=bare-except
+        log.info("User someusr does not exist.")
+
+    return False
+
+
+def is_group_exists(groupname):
```
Comment (pmyjavec) [inline sriracha/helper/manage_user_groups.py]: `group_exists`

File: sriracha/helper/manage_user_groups.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
+import crypt
+import grp
+import pwd
+import spwd
+import subprocess
+import sys
+
+def is_user_exists(username):
```
Comment (pmyjavec) [inline sriracha/helper/manage_user_groups.py]: `user_exists`

File: sriracha/helper/manage_user_groups.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    try:
+        password = spwd.getspnam(username).sp_pwdp
+        return password == crypt.crypt(expected_passwd, password)
+    except:  # pylint: disable=bare-except
+        log.info("Failing to get passsword for the user %s", username)
+
+    return False
+
+
+def check_gid(groupname, expected_gid):
```
Comment (pmyjavec) [inline sriracha/helper/manage_user_groups.py]: This comment and function name is ambiguous, what does it do?

File: sriracha/helper/manage_user_groups.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+        if actual_gid == int(expected_gid):
+            return True
+    except:  # pylint: disable=bare-except
+        log.info("Failed to get group id for %s", groupname)
+
+    return False
+
+
+def user_exists_in_group(groupname, username):
```
Comment (pmyjavec) [inline sriracha/helper/manage_user_groups.py]: `user_in_group`

File: sriracha/helper/manage_user_groups.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+
+def check_gid(groupname, expected_gid):
+    """Checks groupid for groupname"""
+    try:
+        actual_gid = grp.getgrnam(groupname).gr_gid
+
+        if actual_gid == int(expected_gid):
+            return True
+    except:  # pylint: disable=bare-except
```
Comment (pmyjavec) [inline sriracha/helper/manage_user_groups.py]: Fix the bare-except here and always use the correct exception.

File: sriracha/helper/manage_user_groups.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+    return False
+
+
+def user_exists_in_group(groupname, username):
+    """Checks user existence in the group"""
+    try:
+        grp_mems = grp.getgrnam(groupname).gr_mem
+
+        if username in grp_mems:
```
Comment (pmyjavec) [inline sriracha/helper/manage_user_groups.py]: Same here, move this if statement to below the try/except block

File: sriracha/helper/manage_user_groups.py [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+        output = subprocess.run(
+            cmd, check=False, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL
+        )
+
+        if actual_shell in expected_shells and output.returncode == 1:
+            return True
+    except:  # pylint: disable=bare-except
+        log.info("Failed to get the user shell for %s", username)
+
+    return False
```
Comment (pmyjavec) [inline sriracha/helper/manage_user_groups.py]: Please fix this. There should be new lines at the end of the file.

File: sriracha/helper/manage_user_groups.py [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+        if actual_grp_name == groupname:
+            return True
+    except:  # pylint: disable=bare-except
+        log.info("Failed to get user group of %s", groupname)
+
+    return False
+
+
+def check_non_interactive_shell(username):
```
Comment (pmyjavec) [inline sriracha/helper/manage_user_groups.py]: `def user_shell_is_not_interactive`

File: sriracha/helper/manage_user_groups.py [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+
+def check_user_primary_grp(groupname, username):
+    """Checks primary group of the user"""
+    try:
+        p_gid = pwd.getpwnam(username).pw_gid
+        actual_grp_name = grp.getgrgid(p_gid).gr_name
+
+        if actual_grp_name == groupname:
+            return True
+    except:  # pylint: disable=bare-except
```
Comment (pmyjavec) [inline sriracha/helper/manage_user_groups.py]: Same, don't use blanket excepts here. These warnings you're commented out are in here (usually) for good reason.

File: sriracha/helper/manage_user_groups.py [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+    return False
+
+
+def check_user_primary_grp(groupname, username):
+    """Checks primary group of the user"""
+    try:
+        p_gid = pwd.getpwnam(username).pw_gid
+        actual_grp_name = grp.getgrgid(p_gid).gr_name
+
+        if actual_grp_name == groupname:
```
Comment (pmyjavec) [inline sriracha/helper/manage_user_groups.py]: Move to below try/except block

File: sriracha/helper/manage_user_groups.py [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+    try:
+        pwd.getpwnam(username)
+        return True
+    except:  # pylint: disable=bare-except
+        log.info("User someusr does not exist.")
+
+    return False
+
+
+def group_exist(groupname):
```
Comment (pmyjavec) [inline sriracha/helper/manage_user_groups.py]: `group_exists`

File: sriracha/helper/manage_user_groups.py [hunk 12]
```diff
@@ -0,0 +1,98 @@
+import crypt
+import grp
+import pwd
+import spwd
+import subprocess
+import sys
+
+def user_exist(username):
```
Comment (pmyjavec) [inline sriracha/helper/manage_user_groups.py]: `user_exists`

File: sriracha/helper/cmd.py [hunk 13]
```diff
@@ -0,0 +1,8 @@
+import subprocess
+
+def exec_cmd(cmd):
+    try:
+        subprocess.run(cmd, check=True)
+    except:  # pylint: disable=bare-except
+        log.error("Failed to execute the command.")
+        sys.exit(1)
```
Comment (pmyjavec) [inline sriracha/helper/cmd.py]: add new line.

**Review by pmyjavec — CHANGES_REQUESTED** [review]

File: sriracha/helper/manage_user_groups.py (hunk 3 above)
Comment (ramreddya) [reply sriracha/helper/manage_user_groups.py]: It checks the expected `group_id` is set for the `group`.

Comment (pmyjavec) [reply sriracha/helper/manage_user_groups.py]: `groupname_equals_gid`

File: sriracha/helper/manage_user_groups.py [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+    try:
+        grp.getgrnam(groupname)
+        return True
+    except KeyError as e:
+        log.info("Group %s does not exist.", groupname)
+
+    return False
+
+
+def check_passwd(username, expected_passwd):
```
Comment (pmyjavec) [inline sriracha/helper/manage_user_groups.py]: password_is_equal_to:

File: sriracha/helper/manage_user_groups.py [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+        grp_mems = grp.getgrnam(groupname).gr_mem
+    except KeyError as e:
+        log.info("Failed to get group members of the group %s", groupname)
+
+    if username in grp_mems:
+        return True
+    return False
+
+
+def check_user_primary_grp(groupname, username):
```
Comment (pmyjavec) [inline sriracha/helper/manage_user_groups.py]: `check_user_primary_group`

**Review by pmyjavec — CHANGES_REQUESTED** [review]

**Review by pmyjavec — APPROVED** [review]

**Pull request merged**